Makefile

```diff
@@ -15,84 +17,105 @@ secrets:
 		jq -r .SecretString | \
 		python -m json.tool > gcp-credentials-logs-travis.json
 
-.PHONY: rev
-rev:
-	echo $(APPS_REVISION)
 
-# terraform init
-init-%:
-	cd $(subst .,/,$*)/ && $(MAKEFILE_DIR)/scripts/init.sh
+.PHONY: check-terraform
+check-terraform:
+	bash -c 'if [ "$$(cat infrastructure/.terraform/terraform.tfstate | jq -r .backend.config.profile)" != "$(AWS_PROFILE)" ] ; then echo "AWS profile in local terraform state does not match $(AWS_PROFILE)!" && false ; fi'
+
+
+## CLEAN
+.PHONY: clean-terraform
+clean-terraform: clean-terraform-infrastructure
+	for c in $(APPS); do \
+		$(MAKE) clean-terraform-apps.$$c; \
+	done
+
+.PHONY: clean-terraform-%
+clean-terraform-%:
+	cd $(subst .,/,$*)/ && rm -rf .terraform terraform*
 
+
+## INIT
+# terraform init
 .PHONY: init
 init: \
 	init-infrastructure \
-	init-apps.gcp_to_cwl \
-	init-apps.cwl_to_slack \
-	init-apps.es_idx_manager \
-	init-apps.firehose_to_es_processor \
-	init-apps.cwl_firehose_subscriber \
-	init-ci
+	init-apps
 
-clean-terraform-%:
-	cd $(subst .,/,$*)/ && rm -rf .terraform terraform*
+.PHONY: init-apps
+init-apps:
```

Comment:
What you have here is functional, but if you want these rules to be independent, then what you really want is:

```
INIT_APPS := $(addprefix init-, $(APPS))
init-apps: $(INIT_APPS)
init-%:
    xxxxx
.PHONY: init-apps $(INIT_APPS)
```

Reply:
and repeat for init clean etc.